### ISSUE TITLE
Grid3D broken by numpy 1.10

### ISSUE THREAD

**Issue by guyer** [usnistgov/fipy#475]
As of numpy 1.10, any uses of `Grid3D` fail, e.g.
```python
======================================================================
FAIL: _test (fipy.meshes.uniformGrid3D.UniformGrid3D)
Doctest: fipy.meshes.uniformGrid3D.UniformGrid3D._test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/guyer/anaconda/envs/py2_7_debug/lib/python2.7/doctest.py", line 2226, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for fipy.meshes.uniformGrid3D.UniformGrid3D._test
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/uniformGrid3D.py", line 541, in _test

----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/uniformGrid3D.py", line 667, in fipy.meshes.uniformGrid3D.UniformGrid3D._test
Failed example:
    print numerix.allclose(faceToCellDistanceRatios, mesh._faceToCellDistanceRatio, atol = 1e-10, rtol = 1e-10) # doctest: +PROCESSOR_0
Exception raised:
    Traceback (most recent call last):
      File "/Users/guyer/anaconda/envs/py2_7_debug/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest fipy.meshes.uniformGrid3D.UniformGrid3D._test[49]>", line 1, in <module>
        print numerix.allclose(faceToCellDistanceRatios, mesh._faceToCellDistanceRatio, atol = 1e-10, rtol = 1e-10) # doctest: +PROCESSOR_0
      File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/uniformGrid3D.py", line 278, in _faceToCellDistanceRatio
        numerix.ravel(YZdis.swapaxes(0,2))), axis=1)
    IndexError: axis 1 out of bounds [0, 1)
```

The issue is that `numerix.ravel` returns a 1D `array` and so `axis=1` is nonsensical in `_faceToCellDistanceRatio`. It looks like this was always wrong (going back to [at least 2011](https://github.com/usnistgov/fipy/blame/develop/fipy/meshes/uniformGrid3D.py#L276)), but numpy tolerated it until now. 

Probably safe to just remove the `axis=1` argument, but need to test.

**Comment by guyer** [comment]
Fixed by #478 / 33d033d

**Comment by DarioPanada** [comment]
This seems to still be an issue on clean pip installations, tentatively fixed by removing axis=1 parameter on line 279, uniformGrid3D.py
